src/OpenTelemetry.Api/Baggage.cs

```diff
@@ -164,7 +166,7 @@ public static Baggage SetBaggage(string name, string value, Baggage baggage = de
     /// <returns>New <see cref="Baggage"/> containing the new key/value pairs.</returns>
     /// <remarks>Note: The <see cref="Baggage"/> returned will be set as the new <see cref="Current"/> instance.</remarks>
     [SuppressMessage("roslyn", "RS0026", Justification = "TODO: fix APIs that violate the backcompt requirement - multiple overloads with optional parameters: https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md.")]
-    public static Baggage SetBaggage(IEnumerable<KeyValuePair<string, string>> baggageItems, Baggage baggage = default)
+    public static Baggage SetBaggage(IEnumerable<KeyValuePair<string, string?>> baggageItems, Baggage baggage = default)
```

Comment:
`Create` methods does not accept null as a values. It could, as it is filtered out. This method accept. Seems like inconsistency.

Reply:
It could be `IEnumerable<KeyValuePair<string, string?>>?` here. I decided to not do that though because it felt awkward. User is calling `SetBaggage` so presumably they have some `Baggage` and they want to set some items on it. Passing `null` seems odd/not useful for this. Could even be confusing. Maybe the user thinks that passing `null` would clear everything? But it doesn't 😄 We have `ClearBaggage` for that. We also support `Baggage.Create()` to create an empty instance. So ya even while `null` is allowed it felt better to not promote that case through the annotations. Just my opinion happy to be convinced otherwise if you feel strongly about it.

Reply:
I think that I wrongly asked question.
The value of KeyValuePair cannot be null in create:
```
Baggage Create(Dictionary<string, string>? baggageItems = null)
```. Setting allows this.

Lack of nullability of baggeItems here is fine.

Reply:
Gotcha. No problem 😄 

Why is it `Baggage Create(Dictionary<string, string>? baggageItems = null)` and not `Baggage Create(Dictionary<string, string?>? baggageItems = null)`? Kind of the same answer as above, but in reverse!

For setting some baggage items we have `SetBaggage(string key, string? value)` (single item) and `Baggage SetBaggage(IEnumerable<KeyValuePair<string, string?>> baggageItems)` bulk set.

If you pass a `null` value, it is just like calling `RemoveBaggage`.

So we allow our "SetBaggage" operation to remove stuff. Kind of strange perhaps, but it is what it is.

Now looking at the `Create` API. User is creating a brand new baggage instance. The question I asked myself is: Why would you need to remove something from a baggage which hasn't been created yet? Seemed like the "Create" API shouldn't have the same "null-value-means-remove" behavior 🤷

Reply:
The question is if similar API should operate on different types.
Especially when there is no easily casting even from Dictionary<string, string> to Dictionary<string, string>.

I do not have good answer here, Approving. Thanks!